squad/api/rest.py

```diff
@@ -571,6 +572,25 @@ def unsubscribe(self, request, pk=None):
         data = {"email": subscriber_email}
         return Response(data, status=status.HTTP_200_OK)
 
+    @action(detail=True, methods=['get', 'post'], suffix='compare_builds', permission_classes=[AllowAny])
+    def compare_builds(self, request, pk=None):
+        builds_to_compare = {
+            "first": request.data.get("first", None) if request.method == 'POST' else request.query_params.get("first", None),
+            "second": request.data.get("second", None) if request.method == 'POST' else request.query_params.get("second", None)
+        }
+        if any(builds_to_compare.values()) and all(builds_to_compare.values()):
+            try:
+                builds = self.get_object().builds.filter(version__in=builds_to_compare.values()).order_by('-created_at')
```

Comment:
I think this is wrong. You should instantiate the builds separately as the baseline might come after the build to compare. This is a silly usecase but IMHO we should allow it.

Do you expect version strings to be sent to this method? IMHO it would be better to expect IDs.

Reply:
@mwasilew I agree it should be ids not versions. I understand the corner case you talk about but what if users wants to compare two specific builds regardless of a baseline existence? Is there such a case even? or should we only compare the passed baseline arg only not a newer one exists?

Reply:
I don't think it matters which builds you compare. Baseline is 'first' build passed to TestComparison. The code you proposed uses oldest as 'first'. My request it so give user opportunity to select the build which he considers baseline and not make the selection 'secretly'.